src/app/tarea-dialog/tarea-dialog.component.ts

```diff
@@ -15,34 +19,58 @@ export class TareaDialogComponent implements OnInit, OnDestroy {
 
   public formTarea!: FormGroup;
   private subscription!: Subscription;
+  estadoTareas = Object.values(TareaStatus);
 
   constructor(private formBuild: FormBuilder,
     private tareaSvc: TareaService,
     private _snackBar: MatSnackBar,
-    private spinner: SpinnerService) { }
+    private spinner: SpinnerService,
+    public dialog: MatDialog,
+    @Inject(MAT_DIALOG_DATA) public tarea_Edit: Tarea
+  ) { }
 
   ngOnInit(): void {
+
     this.formTarea = this.formBuild.group({
       titulo: ['', [Validators.required, Validators.maxLength(30), Validators.pattern(/^[a-zA-Z 0-9.]+$/)]],
-      description: ['', Validators.maxLength(250)]
-
+      description: ['', Validators.maxLength(250)],
+      status: ['']
     });
 
+    if (this.tarea_Edit) { }
+    this.formTarea.patchValue(this.tarea_Edit);
+
   }
+
   ngOnDestroy(): void {
+
+    if (this.subscription === undefined)
+      return;
+
     this.subscription.unsubscribe();
+
   }
 
-  send(): void {
+  CrearTarea(): void {
     this.spinner.activeSpinner();
     this.subscription = this.tareaSvc.addTarea$(this.formTarea.value).pipe(
       finalize(() => this.spinner.desactiveSpinner()))
       .subscribe({
         next: dat => this._snackBar.open(dat.titulo, "se a añadido...", { duration: 2000 }),
         error: err => this._snackBar.open(err, "", { duration: 2000 }),
-        complete: () => this.formTarea.reset({ titulo: '', description: '' })
+        complete: () => this.formTarea.reset({ titulo: '', description: '', status: '' })
       });
 
   }
 
+  editarTarea(): void {
+    const id_tareaEdit = this.tarea_Edit.id;
```

Comment:
este methodo de editar tarea deberia tener solo una responsabilidad y es la de llamara al methodo de editar en el servicio, pero ademas de eso esta extrayendo el .

yo diria de crear un methodo privado que devuelva la tarea lista para guardarla, algo como esto:
```
private getTareaReadyToSave(oldTareaId:number): Tarea {
   return {
     id: oldTareaId,
     ...this.formTarea.value
   } as Tarea;
}
```

entonces pasar ese method como argumento en this.tareaSvc.editTarea$.